warehouse/query-core/src/main/java/datawave/query/language/parser/jexl/JexlNodeSet.java

```diff
@@ -0,0 +1,185 @@
+package datawave.query.language.parser.jexl;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+import datawave.query.jexl.JexlASTHelper;
+import org.apache.commons.jexl2.parser.JexlNode;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Utility class that implements the {@link Set} interface for use with a collection of Jexl nodes.
+ *
+ * Element uniqueness is computed using {@link JexlASTHelper#nodeToKey(JexlNode)} for keys.
+ *
+ * Important Note: The set is considered invalid if any of the underlying Jexl nodes change.
+ */
+public class JexlNodeSet implements Set<JexlNode> {
+    
+    // Internal map of node keys to nodes;
+    private final Map<String,JexlNode> nodeMap;
+    
+    public JexlNodeSet() {
+        this.nodeMap = new HashMap<>();
+    }
+    
+    /**
+     * Method to access all the Jexl nodes in the set.
+     * 
+     * @return - the underlying node map values.
+     */
+    public Collection<JexlNode> getNodes() {
+        return nodeMap.values();
+    }
+    
+    /**
+     * Get the set of generated node keys.
+     * 
+     * @return - the underlying node map keySet.
+     */
+    public Set<String> getNodeKeys() {
+        return nodeMap.keySet();
+    }
+    
+    @Override
+    public int size() {
+        return nodeMap.size();
+    }
+    
+    @Override
+    public boolean isEmpty() {
+        return nodeMap.isEmpty();
+    }
+    
+    @Override
+    public boolean contains(Object o) {
+        if (o instanceof JexlNode) {
+            String nodeKey = JexlASTHelper.nodeToKey((JexlNode) o);
```

Comment:
This doesn't build--looks like this method was removed [here](https://github.com/NationalSecurityAgency/datawave/commit/d65a509315f42fda7cd9b6f989f8fca4bc348351#diff-dd6995b5d33081727484b35ba74f8b0bL1253).

Reply:
Updated to add that method back in.